gui/web/src/components/atoms/Button/Button.js

```diff
@@ -55,17 +52,27 @@ class Button extends Component {
       return
     }
     
+    if (this.props.eventName === "" || this.props.eventName === undefined) {
+      return
+    }
+    
     var _this = this
-    // TODO DS populate the eventData.
-    this._asyncRequests["sendMetricEvent"] = sendMetricEvent(this.props.baseUrl, this.props.eventName, {}).then(resp => {
+    var eventData = {
```

Comment:
`const` instead of `var` (I should have been more diligent about this earlier to have set a better precedent)

Reply:
PS: we can probably remove the single-quotes around the field, since this is javascript in which json is native